app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt

```diff
@@ -294,6 +294,47 @@ class ParkRepositoryFirestore(private val db: FirebaseFirestore) : ParkRepositor
     }
   }
 
+  /**
+   * Add a rating to a park if the user has not already rated it.
+   *
+   * @param pid The park ID.
+   * @param uid The user ID of the person rating.
+   * @param rating The rating to add from 1 to 5.
+   */
+  override suspend fun addRating(pid: String, uid: String, rating: Float) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
```

Comment:
I might be missing something as the whole viewModel is structured like this and I haven't worked on it but:
What happens when the require fails? I haven't seen the exception being handled here or in the viewModel. Is it supposed to be handled in the UI? Might be worth adding it in the description of the function if that's the case

Reply:
When `require` statement fails, it throws an `IllegalArgumentException` with the message `PID_EMPTY`, preconditions failing means that the developper is not using the function correctly, it's hard for me to see a case where it would be useful for the MVVM to "replace" a bad parameter with a good one without the developper being aware that its use is problematic.

Reply:
fair enough, wasn't sure if the app crashing was the expected behavior